build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -83,22 +86,45 @@
       <!-- CopySharedFramework -->
       <Copy SourceFiles="@(SharedFramework)"
             DestinationFiles="@(SharedFramework->'$(StageDirectory)\%(RecursiveDir)%(Filename)%(Extension)')" />
-    
+
       <!-- Publish DotNet -->
-      <Exec Command="%(Stage.DotnetExe) publish
-                     --native-subdirectory
-                     --output $(SdkOutputDirectory)
-                     --configuration $(Configuration)
-                     --version-suffix $(CommitCount)
-                     $(SrcDirectory)/dotnet" />
+      <DotNetPublish ToolPath="%(Stage.DotnetDir)"
+                     NativeSubdirectory="True"
+                     Output="$(SdkOutputDirectory)"
+                     Configuration="$(Configuration)"
+                     VersionSuffix="$(CommitCount)"
+                     ProjectPath="$(SrcDirectory)/dotnet" />
 
       <!-- Corehostify Binaries -->
-      <FixModeFlags Dir="$(SdkOutputDirectory)" />
+      <ItemGroup Condition=" '$(OSName)' != 'win' ">
+        <SdkOutputChModTargets Remove="*" />
+        <SdkOutputChModTargets Include="$(SdkOutputDirectory)/**/*.exe;
+                                      $(SdkOutputDirectory)/**/*.dll" >
+          <!-- Managed assemblies do not need execute -->
+          <Mode>u=rw,g=r,o=r</Mode>
+        </SdkOutputChModTargets>
+
+        <SdkOutputChModTargets Include="$(SdkOutputDirectory)/**/*.dylib;
+                                      $(SdkOutputDirectory)/**/*.so" >
+          <!-- Generally, dylibs and sos have 'x' -->
+          <Mode>u=rwx,g=rx,o=rx</Mode>
+        </SdkOutputChModTargets>
+
+        <SdkOutputChModTargets Include="$(SdkOutputDirectory)/**/*"
+                             Exclude="$(SdkOutputDirectory)/**/*.*" >
+          <!-- Executables need x -->
+          <Mode>u=rwx,g=rx,o=rx</Mode>
+        </SdkOutputChModTargets>
+      </ItemGroup>
+
+      <ChMod Condition=" '$(OSName)' != 'win' "
```

Comment:
I'm happy to change, but I wanted to express `change mode` as described [here](http://ss64.com/bash/chmod.html).